api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -665,6 +665,11 @@ message HttpConnectionManager {
   // information about internal/external addresses.
   InternalAddressConfig internal_address_config = 25;
 
+  // If set, Envoy will not append the
+  // :ref:`config_http_conn_man_headers_x-envoy-local-overloaded` HTTP header in the scenario where
+  // the Overload Manager has been triggered. This defaults to true.
+  bool skip_local_overload_append = 57;
```

Comment:
Sorry, I may be missing something, but want to make a few comments on the API:
- Try to avoid a bool-value that defaults to true. The reason behind this is that upgrading an Envoy without upgrading the config-server will change the behavior by default, and make upgrading Envoy more challenging. It is better to use an opt-in (defaults to false) field rather than an opt-out.
- Another case for opt-in is if this is changing the behavior of the proxy. Adding a new header without explicitly setting a config-knob may have non-desirable outcome (e.g., the number of headers grows by 1, and an overflow occurs).

Reply:
Thank you. I did not know about the upgrade scenario.
I will update the variable to still be opt in but this time with "false" as default value :+1: